rtl/eclrtl/rtlrecord.cpp

```diff
@@ -300,20 +401,28 @@ size32_t RtlRecord::getMinRecordSize() const
 
     size32_t minSize = 0;
     for (unsigned i=0; i < numFields; i++)
-        minSize += queryType(i)->getMinSize();
-
+    {
+        const RtlFieldInfo *field = fields[i];
+        if (!(field->flags & RFTMinifblock))
+            minSize += queryType(i)->getMinSize();
+    }
     return minSize;
 }
 
 size32_t RtlRecord::deserialize(ARowBuilder & rowBuilder, IRowDeserializerSource & in) const
 {
     size32_t offset = 0;
+    byte *conditionValues = (byte *) alloca(numIfBlocks);
     for (unsigned i = 0; i < numVarFields; i++)
     {
+        memset(conditionValues, 2, numIfBlocks);
```

Comment:
Should this be outside the loop?

Reply:
It should be once per row. I guess it still works inside the loop but rather less efficient!